activesupport/lib/active_support/core_ext/name_error.rb

```diff
@@ -1,6 +1,13 @@
 # frozen_string_literal: true
 
 class NameError
+  attr_writer :receiver
+  alias_method :original_receiver, :receiver
+
+  def receiver
+    @receiver ||= original_receiver
```

Comment:
So MRI store the `receiver` in a private instance variable (`receiver` with no `@`), there is no way to set it from Ruby-land.